Source/CMakeLists.txt

```diff
@@ -19,7 +19,10 @@ else()
   # Enable C++17, but fall back to C++14 if it isn't available.
   # CMAKE_CXX_STANDARD cannot be used here because we require C++14 or newer, not any standard.
   check_and_add_flag(CXX17 -std=c++17)
-  if(NOT FLAG_CXX_CXX17)
+  if(FLAG_CXX_CXX17)
+    # clang 6 complains about the 'register' keyword in included Xlib headers
+    check_and_add_flag(REGISTER -Wno-register)
```

Comment:
This suppression should likely be applied within the file causing the warning if possible (preferably just around the culprit includes only). We'd likely still want to warn about it if any code using `register` is attempted to be added to the codebase itself.

If that's not reasonably doable, then LGTM.

Reply:
Good idea, changed to clang pragmas for that file only